deployment/ccip/changeset/v1_6/cs_deploy_usdc_token_pools.go

```diff
@@ -0,0 +1,174 @@
+package v1_6
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-evm/pkg/utils"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/latest/mock_usdc_token_messenger"
+	// TODO: New token pool contract should be imported from the latest version
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/latest/usdc_token_pool"
+	"github.com/smartcontractkit/chainlink-evm/gethwrappers/shared/generated/erc20"
+
+	cldf_evm "github.com/smartcontractkit/chainlink-deployments-framework/chain/evm"
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview/evm"
+)
+
+var (
+	DeployUSDCTokenPoolNew = cldf.CreateChangeSet(deployUSDCTokenPoolContractsLogic, deployUSDCTokenPoolContractsPrecondition)
+)
+
+// DeployUSDCTokenPoolInput defines all information required of the user to deploy a new USDC token pool contract.
+type DeployUSDCTokenPoolInput struct {
+	// PreviousPoolAddress is the address of the previous USDC token pool contract, inflight messages
+	// are redirected to the previous pool when needed.
+	PreviousPoolAddress common.Address
+	// TokenMessenger is the address of the USDC token messenger contract.
+	TokenMessenger common.Address
+	// USDCTokenAddress is the address of the USDC token for which we are deploying a token pool.
+	TokenAddress common.Address
+	// AllowList is the optional list of addresses permitted to initiate a token transfer.
+	// If omitted, all addresses will be permitted to transfer the token.
+	AllowList []common.Address
+}
+
+func (i DeployUSDCTokenPoolInput) Validate(ctx context.Context, chain cldf_evm.Chain, state evm.CCIPChainState) error {
+	// Ensure that required fields are populated
+	if i.TokenAddress == utils.ZeroAddress {
+		return errors.New("token address must be defined")
+	}
+	if i.TokenMessenger == utils.ZeroAddress {
+		return errors.New("token messenger must be defined")
+	}
+	if _, ok := state.CCTPMessageTransmitterProxies[deployment.Version1_6_0]; !ok {
+		// TODO: Could this be deployed automatically if it doesn't exist?
+		return fmt.Errorf("CCTP message transmitter proxy for version %s not found on %s", deployment.Version1_6_0, chain)
+	}
+	if i.PreviousPoolAddress == utils.ZeroAddress {
+		return errors.New("previous pool address must be defined")
+	}
```

Comment:
I guess it's possible that there isn't a previous token pool. Maybe there should be a "NoPreviousTokenPool" flag to ensure this field isn't skipped by accident.